apps/mark/frontend/src/pages/poll_worker_screen.tsx

```diff
@@ -495,18 +509,18 @@ export function PollWorkerScreen({
   const [isDiagnosticsScreenOpen, setIsDiagnosticsScreenOpen] = useState(false);
   const [scannerReportData, setScannerReportData] =
     useState<ScannerReportData>();
+
   useEffect(() => {
-    if (scannerReportData) return;
-    if (!pollworkerAuth.card.hasStoredData) return;
-    async function loadTallyFromCard() {
-      setScannerReportData(
-        (
-          await pollworkerAuth.card.readStoredObject(ScannerReportDataSchema)
-        ).ok()
-      );
-    }
-    void loadTallyFromCard();
-  }, [pollworkerAuth, scannerReportData]);
+    readScannerReportDataFromCardMutation.mutate(undefined, {
+      onSuccess(result) {
+        if (result.isOk() && result.ok() !== undefined) {
+          setScannerReportData(result.ok());
+        }
+      },
+    });
+    // Try to read scanner report data from the card once on mount
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
```

Comment:
Only performing this action on mount has conceptually always been what we wanted. When I actually populated the dependency array fully, `readScannerReportDataFromCardMutation` caused the dreaded `useEffect` infinite loop because it's redefined on every render. I didn't want to use a `useCallback` as a bandaid to work around that and instead wanted to be more explicit.

I spent some time thinking about how to move this out of a `useEffect` altogether and into an event handler, but what makes this challenging is that this read isn't triggered by a button click - it's triggered by card insertion.

Overall, I'm okay with where this landed!

Reply:
How about using `useQueryChangeListener` to detect when  `authStatusQuery` changes to a logged in pollworker? (It would have to be moved up to a parent component, but that might not be the worst thing in the world...)

Reply:
Ah nice great suggestion! Updated ✅